pkg/apis/metrics/types.go

```diff
@@ -16,14 +16,427 @@ limitations under the License.
 
 package metrics
 
-import "k8s.io/kubernetes/pkg/api/unversioned"
+import (
+	"time"
 
-// Placeholder top-level node resource metrics.
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RawNode holds node-level unprocessed sample metrics.
 type RawNode struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Node.
+	NodeRef api.ObjectReference `json:"nodeRef,omitempty"`
+	// Overall machine metrics.
+	Machine RawContainer `json:"machine,omitempty"`
+	// Metrics of system components.
+	SystemContainers []RawContainer `json:"systemContainers,omitempty"`
 }
 
-// Placeholder top-level pod resource metrics.
+// RawPod holds pod-level unprocessed sample metrics.
 type RawPod struct {
 	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata, applying to the lists of stats.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:",inline"`
+	// Reference to the measured Pod.
+	PodRef api.ObjectReference `json:"podRef,omitempty"`
+	// Metrics of containers in the measured pod.
+	Containers []RawContainer `json:"containers,omitempty"`
+}
+
+// RawContainer holds container-level unprocessed sample metrics.
+type RawContainer struct {
+	// Reference to the measured container.
+	Name string `json:"name,omitempty"`
+	// Describes the container resources.
+	Spec ContainerSpec `json:"spec,omitempty"`
```

Comment:
Is this really a Spec, in the sense we normally use that term? I.e., would a user POST/PUT/PATCH and change or set this data? If not, I think this is actually Status, not Spec.